model/configuration.py

```diff
@@ -769,3 +774,567 @@ def excluded_audio_data_sources(cls, _db):
         if value is None:
             value = cls.EXCLUDED_AUDIO_DATA_SOURCES_DEFAULT
         return value
+
+
+class ExternalIntegrationOwner(object):
+    """Interface allowing to get access to an external integration"""
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def external_integration(self, db):
+        """Returns an external integration owned by this object
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :return: External integration owned by this object
+        :rtype: core.model.configuration.ExternalIntegration
+        """
+        raise NotImplementedError()
+
+
+class CollectionExternalIntegrationOwner(object):
+    """Interface allowing to get access to an external integration"""
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def collection_external_integration(self, db, collection_id):
```

Comment:
Our general rule is that, whenever possible, we pass around data model objects rather than database session + row ID. You can always get the database session from a data model object with `Session.object_session(obj)`.

Reply:
Changed signature to the following:
```
def collection_external_integration(self, collection):
```